core/templates/components/question-directives/questions-list/questions-list.component.ts

```diff
@@ -425,13 +426,33 @@ angular.module('oppia').component('questionsList', {
         }
         $location.hash(ctrl.questionId);
       };
-      ctrl.deleteQuestionFromSkill = function(
+
+      ctrl.deleteQuestionFromSkill = async function(
           questionId, skillDescription) {
         if (!ctrl.canEditQuestion()) {
           AlertsService.addWarning(
             'User does not have enough rights to delete the question');
           return;
         }
+
+        let skillIsAssgignedForDiagnosticTest = false;
+        await TopicsAndSkillsDashboardBackendApiService
+          .fetchTopicAssignmentsForSkillAsync(
+            ctrl.selectedSkillId).then((response: AssignedSkill[]) => {
+            response.map((topic) => {
+              if (topic.skillIdsForDiagnosticTest.includes(
+                ctrl.selectedSkillId)) {
+                skillIsAssgignedForDiagnosticTest = true;
+              }
+            });
+          });
+
```

Comment:
The below two conditions are satisfied --  don't allow for question deletion.

The constant 2 in the second check is a condition that at least 2 questions should be present in a skill for getting assigned for the diagnostic test.

Reply:
Extract 2 into a named constant, otherwise readers of this file won't understand what it represents conceptually.

Perhaps also add a comment above this block to explain the rationale behind the constraint.